File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;

        if (!s.startsWith("(")) {
            return false;
        }
        
        Queue<Character> queue = new ArrayDeque<>();        
        queue.add(s.charAt(0));
        
        for (int i = 1; i<s.length(); i++) {
            char c = s.charAt(i);
            if (c == '(') {
                queue.add(c);
                continue;
            }
            queue.poll();
                
        }
        if (queue.isEmpty()){
            return true;
        }        
        return false;

    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    count = 0
    for i,v in enumerate(s):
        if i == 0 and v != '(' or count < 0:
            return False
        elif v == '(':
            count += 1
            continue
        elif v == ')':
            count -= 1
    
    if count == 0: return True
    else: return False